### PR TITLE
Add CET IBT support in library crates

### PR DESCRIPTION
**Pull request by gaojiaqi7**
Need to add `endbr64` in front of the position where the global asm functions starts.

- Replace the global asm functions in `td-exception` with the inline functions provided by crate `x86_64`
- Add `endbr64` in the global asm funcitons in `tdx-tdcall`

### CODE REVIEW

**Comment by gaojiaqi7** [comment]
IBT support is tracked by the issue https://github.com/confidential-containers/td-shim/issues/306

File: tdx-tdcall/src/asm/tdcall.asm [hunk 1]
```diff
@@ -19,6 +19,7 @@
 # )
 .global asm_td_call
 asm_td_call:
+        endbr64
```
Comment (jyao1) [inline tdx-tdcall/src/asm/tdcall.asm:22]: why we need endbr here?

Comment (gaojiaqi7) [reply tdx-tdcall/src/asm/tdcall.asm:22]: It seems global asm functions are called with an indirect CALL/JMP, so the program will crash if we do not add `endbr64` here.

Comment (jyao1) [reply tdx-tdcall/src/asm/tdcall.asm:22]: Would you please add some reference on that? I am a little surprised on why.

Comment (gaojiaqi7) [reply tdx-tdcall/src/asm/tdcall.asm:22]: This is the disassembly of https://github.com/confidential-containers/td-shim/blob/main/tdx-tdcall/src/lib.rs#L101-L103

```
0000000000000000 <td_call>:
   0:	f3 0f 1e fa          	endbr64 
   4:	48 83 ec 28          	sub    $0x28,%rsp
   8:	48 89 f9             	mov    %rdi,%rcx
   b:	ff 15 00 00 00 00    	call   *0x0(%rip)        # 11 <td_call+0x11>
  11:	48 83 c4 28          	add    $0x28,%rsp
  15:	c3                   	ret
```

Comment (jyao1) [reply tdx-tdcall/src/asm/tdcall.asm:22]: OK. It is indirect call.

**Review by jyao1 — APPROVED** [review]

**Comment by jyao1** [comment]
please resolve conflict.

**Pull request merged**